support/iphone/main.m

```diff
@@ -50,23 +47,20 @@ int main(int argc, char *argv[])
   [[TiLogServer defaultLogServer] setPort:TI_LOG_SERVER_PORT];
 #endif
 
-  UIColor *defaultBgColor = UIColor.blackColor;
-#if defined(DEFAULT_BGCOLOR_RED) && defined(DEFAULT_BGCOLOR_GREEN) && defined(DEFAULT_BGCOLOR_BLUE)
-  defaultBgColor = [UIColor colorWithRed:DEFAULT_BGCOLOR_RED
-                                green:DEFAULT_BGCOLOR_GREEN
-                                 blue:DEFAULT_BGCOLOR_BLUE
-                                alpha:1.0f];
-#endif
+  UIColor *defaultBgColor = [UIColor colorWithRed:TI_APPLICATION_DEFAULT_BGCOLOR_RED
+                                            green:TI_APPLICATION_DEFAULT_BGCOLOR_GREEN
+                                             blue:TI_APPLICATION_DEFAULT_BGCOLOR_BLUE
+                                            alpha:1.0f];
   [[TiSharedConfig defaultConfig] setDefaultBackgroundColor:defaultBgColor];
 
 #if defined(DEBUG) || defined(DEVELOPER)
   [[TiSharedConfig defaultConfig] setDebugEnabled:YES];
 #endif
 
-#ifdef __LOG__ID__
+#ifdef LOGTOFILE
   NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
   NSString *documentsDirectory = [paths objectAtIndex:0];
-  NSString *logPath = [documentsDirectory stringByAppendingPathComponent:[NSString stringWithFormat:@"%s.log", STRING(__LOG__ID__)]];
+  NSString *logPath = [documentsDirectory stringByAppendingPathComponent:[NSString stringWithFormat:@"%@.log", TI_APPLICATION_GUID]];
```

Comment:
Was there a case where this caused issues?

Reply:
No, but as I wrote (see *Note* above), the intention here is to replace all key-value compiler flags.
The only use of `__LOG__ID__` is in `support/iPhone/main.m` and
> The `__LOG__ID__` flag is not needed because it has exactly the same value as `TI_APPLICATION_GUID`

Reply:
> The intention of the flag (initially used for the logger from @cb1kenobi) IIRC was to be able to attach different loggers if wanted

Thanks for the clarification.

So I added a separate constant `TI_LOG_ID`. The build process filled then the placeholder `__LOG_ID__` with `tiapp.guid` like before &ndash; here one can use another ID instead of the app's GUID.